_docs-sources/pipelines/overview/configuration.md

```diff
@@ -1,87 +1,360 @@
-  is-delegated-repo: false
-  # Only set to true for infrastructure-live-access-control
-  is-access-control-repo: false
-
-  account-vending:
-    sdlc:
-    # Map of account identifiers to account vending configurations.
-    # A valid account vending configuration includes:
-      # account-identifiers: List of account identifiers that will be created (required).
-      account-identifiers:
-        # Alphanumeric account identifiers only. On account requests, an account
-        # will be created for each specified identifier & the account name will
-        # include the identifier. e.g. "<ACCOUNT-FAMILY>-dev"
-        - dev
-        - stage
-        - prod
-      # catalog-repositories: List of repositories used by Terragrunt catalog (optional).
-      catalog-repositories:
-        # List of repositories that contain infrastructure modules that can be easily leveraged as a catalog by delegated repositories vended by the infrastructure-root repository.
-        # For more information, see https://terragrunt.gruntwork.io/docs/features/catalog/
-        - "{{ .RepoBaseUrl }}/{{ .InfraModulesRepoName }}"
-      # github-collaborators:
-      # List of GitHub teams and their permissions automatically added to delegated repositories vended by the infrastructure-root repository.
-      # Valid permissions are: pull, triage, push, maintain and admin (in addition to custom roles if any exist)
-      # See https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization#permissions-for-each-role
-      # for more information on GitHub repository roles.
-      #   - team: 'team-name'
-      #     permission: pull
-    sandbox:
-      account-identifiers:
-        - sandbox
-      catalog-repositories:
-        - "{{ .RepoBaseUrl }}/{{ .InfraModulesRepoName }}"
+
+Placing this configuration in a `gruntwork.hcl` file in the same directory as a `terragrunt.hcl` file will cause Pipelines to assume the `role-to-assume-for-plans` role in the AWS account with ID `an-aws-account-id` when running Terragrunt plan commands, using [OIDC](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services) to authenticate to AWS and assume that role.
+
+In most circumstances, the same role would be assumed by multiple units of configuration within a repository (e.g. all units within a given directory configure resources for the same AWS account). In that circumstance, it would be more appropriate not to define a `gruntwork.hcl` file at all, and instead define a `.gruntwork` directory at the root of the repository with a file ending in `.hcl` that contains configurations that are applicable to all units within that directory.
+
+e.g.
+
+```hcl
+# .gruntwork/environments.hcl
+environment "an_account" {
+  filter {
+    paths = ["an-account/*"]
+  }
```

Comment:
Environments are not equivalent to accounts right? If yes, we should carefully consider the use of "an-account" here. It may be a reader's first engagement with our "environments" declaration and may form a mental model that environments **are** accounts.